server/routes/bookings/bookingDetailsController.ts

```diff
@@ -1,21 +1,32 @@
 import type { RequestHandler } from 'express'
 import { ValidationChain, matchedData, param, validationResult } from 'express-validator'
-import { VisitService } from '../../services'
+import { BookerService, VisitService } from '../../services'
 import getPrisonInformation from '../../constants/prisonInformation'
 import paths from '../../constants/paths'
 
 export default class BookingDetailsController {
-  public constructor(private readonly visitService: VisitService) {}
+  public constructor(
+    private readonly visitService: VisitService,
+    private readonly bookerService: BookerService,
+  ) {}
 
   public view(): RequestHandler {
     return async (req, res) => {
-      const { booker, bookings } = req.session
+      const { booker } = req.session
 
       const errors = validationResult(req)
       if (!errors.isEmpty()) {
         return res.redirect(paths.BOOKINGS.HOME)
       }
 
+      req.session.bookings = req.session.bookings
+        ? req.session.bookings
+        : await this.visitService.getFuturePublicVisits(booker.reference)
+
+      const prisoner = booker.prisoners ? booker.prisoners : await this.bookerService.getPrisoners(booker.reference)
+
+      const { bookings } = req.session
```

Comment:
```suggestion
      const { booker, bookings } = req.session

      const errors = validationResult(req)
      if (!errors.isEmpty() || !bookings?.length) {
        return res.redirect(paths.BOOKINGS.HOME)
      }
```

Something like this suggestion (caveat - untested!) and then keeping the rest as it was might be preferable because:
* simpler - avoids duplicating the code to fetch future bookings
* less potential for confusion because if your session expires on a visit details page you just get returned to the listing. If you went back to, say, `.../1` then in the meantime the another visit could have been booked so your page would reload with info on a **different** visit